types/struct/struct.ts

```diff
@@ -26,34 +32,39 @@ export class Struct<
     this.byteAlign = byteAlign;
   }
 
-  read(dataView: DataView, byteOffset = 0): V {
+  read(dataView: DataView, options: TypeOptions = {}): V {
+    options.byteOffset ??= 0;
     const keys = Object.keys(this.typeRecord);
     const len = keys.length;
     const object = {} as V;
 
     for (let i = 0; i < len; i++) {
       const k = keys[i];
       const [offset, type] = this.typeRecord[k];
+      options.byteOffset += offset;
       object[k as keyof V] = type.read(
         dataView,
-        byteOffset + offset,
+        options,
       ) as V[keyof V];
     }
 
     return object;
   }
 
-  write(value: V, dataView: DataView, byteOffset = 0) {
+  write(value: V, dataView: DataView, options: TypeOptions = {}) {
+    options.byteOffset ??= 0;
     const keys = Object.keys(this.typeRecord);
     const len = keys.length;
     for (let i = 0; i < len; i++) {
       const key = keys[i];
       const [offset, type] = this.typeRecord[key];
-      type.write(value[key], dataView, byteOffset + offset);
+      options.byteOffset += +offset;
```

Comment:
```suggestion
      options.byteOffset += offset;
```
Is offset not always an integer key?

Reply:
Just a typo by me :sweat_smile: